5_databases/week1/homework1.sql

```diff
@@ -0,0 +1,69 @@
+-- Active: 1706443591837@@127.0.0.1@3307@hyf
+-- Active: 1706443591837@@127.0.0.1@3307@hyf
+
+--Find out how many tasks are in the task table
+SELECT COUNT(id) FROM task;
+
+--Find out how many tasks in the task table do not have a valid due date
+DESCRIBE task;
+
+SELECT COUNT(*) FROM task WHERE due_date IS NULL;
+--Find all the tasks that are marked as done
+SELECT task.*
+FROM task
+    JOIN status ON task.status_id = status.id
+WHERE
+    status.name = "Done";
+
+SELECT
+    t.id AS TaskID,
+    t.title AS Title,
+    t.description AS Description,
+    t.created AS CreatedDate,
+    t.due_date AS DueDate,
+    s.name AS Status
+FROM task AS t
+    JOIN status AS s ON t.status_id = s.id
+WHERE
+    s.name = 'Done';
```

Comment:
However this is a valid syntax (s.name, t.id, etc), I don't this that you should use it in a real situation. Imagine you have 10-15 different tables, and it's just hard to remember each of them. The names are already short and simple but they give you an understanding of the content inside but just a letter - don't. You can leave you code as it is but keep in mind my advice for the future😉